### DISCUSSION TITLE
COLR v1 support in wezterm is powered by harfbuzz

### DISCUSSION THREAD

**Discussion by wez** [harfbuzz/harfbuzz#4382] [Show and tell] [answered]
wezterm nightly builds are now using harfbuzz to drive COLR v1 glyph rasterization (via a stripped down cairo+pixman)!
There's also an option to use harfbuzz+cairo to rasterize all outline glyphs, which looks decent but I think freetype still has the edge.

Here's wezterm's character selector showing Noto Color Emoji's COLR v1 glyphs:
![image](https://github.com/harfbuzz/harfbuzz/assets/117777/73410524-5a17-4caa-861f-6afa86d2ea5a)

I haven't hooked up sweep gradients yet. I can see from harfbuzz's tests that there are some fonts with a couple of test glyphs. My visual testing infrastructure is built around resolving a codepoint to a glyph. Is there a simple way to transform `test_glyphs-glyf_colr_1.ttf` so that I can assign eg: `a`, `b`, `c` and so on codepoints to those test glyphs?

**Comment by rsheeter** [comment, marked as answer]
Cool to see!

> Is there a simple way to transform test_glyphs-glyf_colr_1.ttf so that I can assign eg: a, b, c and so on codepoints to those test glyphs?

I believe https://github.com/googlefonts/color-fonts is the source for those test fonts. At a glance (`ttx -o - -t cmap fonts/test_glyphs-glyf_colr_1.ttf`) it looks like they do have codepoints associated. If you don't like those codepoints perhaps you could ttx the font and edit (or add to, multiple mappings to the same glyph is fine) the cmap to fit your needs?

If you dislike editing via ttx a wee python script should work, akin to https://rsheeter.github.io/font101/#ttfont but modifying `font["cmap"]`.

**Reply by wez** [reply]
Ah, thanks for that; somehow I hadn't run into `ttx` yet! Very useful for this sort of thing!

**Reply by wez** [reply]
Related: wezterm bundles an older bitmap based version of Noto Color Emoji. I'd like to update that to the more recent version that has COLRv1 data.  The SVG data that is also present inn that font is not used in wezterm, and dominates the file size. Is it valid for me to simply remove the SVG table when I update my bundled copy? I believe so, but I don't know if there is some nuance! 

The size is a significant factor:

* Bitmap version: 9.5M
* SVG+COLR version: 23M (more than 2x the bitmap size!)
* no-SVG version: 4.6M (I deleted the `<SVG/>` portion of the `.ttx` and regenerated it)

**Reply by anthrotype** [reply]
> Is it valid for me to simply remove the SVG table when I update my bundled copy?

definitely, yes

**Reply by wez** [reply]
Thanks!

**Reply by rsheeter** [reply]
https://github.com/googlefonts/noto-emoji `fonts/Noto-COLRv1.ttf` is the COLRv1 only (no svg or bitmap) version.
